packages/aws-cdk-lib/aws-s3-deployment/test/bucket-deployment.test.ts

```diff
@@ -1500,6 +1500,86 @@ test('if any source has markers then all sources have markers', () => {
   });
 });
 
+test('DeployTimeSubstitutedFile can be used to add substitutions in a file', () => {
```

Comment:
We also need a test case for a variable that appears in the file but not in the substitutions map.